docs/machine-learning/tutorials/taxi-fare.md

```diff
@@ -116,37 +122,37 @@ public static PredictionModel<TaxiTrip, TaxiTripFarePrediction> Train()
 
-Next, load your data into the pipeline. Point to the `_datapath` created initially and specify the delimiter of the .csv file (,). Add the following code into the `Train()` method underneath the last step:
+The first step that the learning pipeline performs is loading data from the training data set. In our case, training data set is stored in the text file, which path is defined by the value of the `_datapath` constant. That file contains the header with the column names, so the first row should be ignored while loading data. Columns in the file are separated by the comma (","). Add the following code into the `Train` method:
 
 ```csharp
-pipeline.Add(new TextLoader(_datapath).CreateFrom<TaxiTrip>(separator:','));
+pipeline.Add(new TextLoader(_datapath).CreateFrom<TaxiTrip>(useHeader: true, separator: ','));
 ```
 
-You'll refer to the columns without the underscores in the code you're creating. Copy the `FareAmount` column into a new column called "Label" using the `ColumnCopier()` function. This column is the **Label**.
+In the next steps we refer to the columns by the names defined in the `TaxiTrip` class.
+
+When the model is trained, the values in the column named **Label** are considered as values to be predicted. As we want to predict the taxi trip fare, copy the `FareAmount` column into the **Label** column. To do that, use <xref:Microsoft.ML.Transforms.ColumnCopier> and add the following code:
 
 ```csharp
 pipeline.Add(new ColumnCopier(("FareAmount", "Label")));
 ```
 
-Conduct some **feature engineering** to transform the data so that it can be used effectively for machine learning. The algorithm that trains the model requires **numeric** features, you transform the categorical data (`VendorId`, `RateCode`, and `PaymentType`) into numbers. The `CategoricalOneHotVectorizer()` function assigns a numeric key to the values in each of these columns. Transform your data by adding this code:
+The algorithm that trains the model requires **numeric** features, so you have to transform the categorical data (`VendorId`, `RateCode`, and `PaymentType`) values into numbers. To do that, use <xref:Microsoft.ML.Transforms.CategoricalOneHotVectorizer>, which assigns different numeric key values to the different values in each of the columns, and add the following code:
 
 ```csharp
 pipeline.Add(new CategoricalOneHotVectorizer("VendorId",
                                              "RateCode",
                                              "PaymentType"));
 ```
 
-The last step in data preparation combines all of your **features** into one vector using the `ColumnConcatenator()` function. This necessary step helps the algorithm easily process your features. Add the following code:
+The last step in data preparation combines all of the **features** into one vector using the <xref:Microsoft.ML.Transforms.ColumnConcatenator> transformation class. This necessary step helps the algorithm to easily process the features. Add the following code:
```

Comment:
It's not really that it helps the algorithm to easily process the features, but rather that the learner looks for the "Features" column by default and only uses that column as the features.